src/main/java/codechicken/nei/HeldItemHandler.java

```diff
@@ -0,0 +1,119 @@
+package codechicken.nei;
+
+import static codechicken.nei.NEIClientUtils.translate;
+
+import java.util.List;
+
+import net.minecraft.client.gui.inventory.GuiContainer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.EnumChatFormatting;
+
+import codechicken.nei.guihook.GuiContainerManager;
+import codechicken.nei.guihook.IContainerInputHandler;
+import codechicken.nei.guihook.IContainerTooltipHandler;
+
+public class HeldItemHandler implements IContainerInputHandler, IContainerTooltipHandler {
+
+    protected HeldItemHandler() {
+
+    }
+
+    public static void load() {
+        HeldItemHandler instance = new HeldItemHandler();
+        GuiContainerManager.addInputHandler(instance);
+        GuiContainerManager.addTooltipHandler(instance);
+    }
+
+    protected boolean contains(int mousex, int mousey) {
+        return NEIClientUtils.getHeldItem() != null
+                && (ItemPanels.bookmarkPanel.contains(mousex, mousey) || ItemPanels.itemPanel.contains(mousex, mousey)
+                        || ItemPanels.itemPanel.historyPanel.contains(mousex, mousey));
+    }
+
+    @Override
+    public boolean mouseClicked(GuiContainer gui, int mousex, int mousey, int button) {
+
+        if (contains(mousex, mousey)) {
+
+            if (NEIClientConfig.canPerformAction("delete") && NEIClientConfig.canPerformAction("item")) {
```

Comment:
Shouldn't this check for creative mode for deletion, as mentioned in your PR? Right now the tooltip for this is showing up in survival mode (I didn't test the actual deletion, because I want my items)